pkg/deploytest/deployment.go

```diff
@@ -103,45 +93,35 @@ func NewDeployment(conf *TestConfig) (*Deployment, error) {
 		logger = logging.Synchronize(logging.ConsoleDebugLogger)
 	}
 
-	// Create a simulated network transport to route messages between replicas.
-	fakeTransport := NewFakeTransport(conf.NumReplicas)
-
 	// Create a dummy static membership with replica IDs from 0 to len(replicas) - 1
-	membership := make([]t.NodeID, conf.NumReplicas)
-	for i := 0; i < len(membership); i++ {
-		membership[i] = t.NewNodeIDFromInt(i)
+	nodeIDs := make([]t.NodeID, conf.NumReplicas)
+	for i := 0; i < len(nodeIDs); i++ {
+		nodeIDs[i] = t.NewNodeIDFromInt(i)
+	}
+
+	// Create a simulated network transport to route messages between replicas.
+	var transportLayer LocalTransport
+	switch conf.Transport {
+	case "fake":
+		transportLayer = NewFakeTransport(nodeIDs)
+	case "grpc":
+		transportLayer = NewLocalGrpcTransport(nodeIDs, logger)
+	case "libp2p":
+		transportLayer = NewLocalLibp2pTransport(nodeIDs, logger)
 	}
 
 	// Create all TestReplicas for this deployment.
 	replicas := make([]*TestReplica, conf.NumReplicas)
 	for i := range replicas {
+		nodeID := t.NewNodeIDFromInt(i)
 
 		// Configure the test replica's node.
 		config := &mir.NodeConfig{
```

Comment:
It would be great to not use & here and use it below in the place where `config` is passed

Reply:
I don't think it is related to generalizing the transport layer interface